agent/taskresource/cgroup/cgroup.go

```diff
@@ -63,14 +65,24 @@ func NewCgroupResource(taskARN string,
 	cgroupRoot string,
 	cgroupMountPath string,
 	resourceSpec specs.LinuxResources) *CgroupResource {
-	return &CgroupResource{
+	c := &CgroupResource{
 		taskARN:         taskARN,
 		control:         control,
 		cgroupRoot:      cgroupRoot,
 		cgroupMountPath: cgroupMountPath,
 		resourceSpec:    resourceSpec,
 		ioutil:          ioutilwrapper.NewIOUtil(),
 	}
+	c.initializeResourceStatusToTransitionFunction()
+	return c
+}
+
+func (c *CgroupResource) initializeResourceStatusToTransitionFunction() {
+	resourceStatusToTransitionFunction := map[taskresource.ResourceStatus]func() error{
+		taskresource.ResourceStatus(CgroupCreated): c.Create,
+		taskresource.ResourceStatus(CgroupRemoved): c.Cleanup,
```

Comment:
So this is currently not useful, right? As we don't use the transition function for cleanup.